lib/Backend/Commands.hs

```diff
@@ -269,6 +269,19 @@ copyCmd (CopyOptions oldPath newPath force) = do
       Left entry -> copyEntry entry
       Right dir -> copyDir dir
 
+  -- Checks that the paths we're trying to copy entries to aren't already occupied
+  -- by existing entries in theirs parent directories.
+  checkCopyOperations operations
+    do \(CopyOperation _ new) -> do
+        let entryPath = new ^. E.path
+        let mPathInits = traverse NE.nonEmpty $ (init . NE.tail . NE.inits) (entryPath ^. pathSegments)
+        case mPathInits of
+          Just pathInits -> do
+            let entryPaths = view (from (pathSegments @EntryPath)) <$> pathInits
+            anyM pathIsEntry entryPaths
```

Comment:
In #34, where we do a check similar to this, we report the parent directory(ies) where an entry was found. We should do the same here.

In fact, now that I think about it, the checks performed in `createCmd` and `copyCmd` are the same. Conceptually, it makes sense: both operations involve creating a new entry.

So it follows that maybe we should have a single function that performs all 3 checks to make sure an entry can be created. And that function could be reused in both `createCmd` and `copyCmd`.

(BTW, this PR should probably be based on top of #34 because you'll need to reuse some stuff from there)

Reply:
Done

Reply:
> So it follows that maybe we should have a single function that performs all 3 checks to make sure an entry can be created. And that function could be reused in both createCmd and copyCmd.

I don't think this part is done yet.

Reply:
I don't know how to do it. We want this function to return smth like `Sem r CheckResult` in order not to lose the result of the check. But in `copyCmd` it would be difficult to deal with this list.

Reply:
Done